api/ombpdf/semtree.py

```diff
@@ -212,8 +210,10 @@ def process_footnotes(self):
                 anno = line.annotation
                 if anno != curr_footnote:
                     curr_footnote = anno
-                    self.writer.create_element(Footnote(anno.number,
-                                                        anno.text))
+                    self.close_all_blocks()
+                    self.open_new_block(Footnote(anno.number), anno)
+                self.process_line(line)
+            self.close_all_blocks()
```

Comment:
Just flagging this line as a departure from what happened before (when there was no need to close any block elements as none were introduced). Is there any situation where closing all of the blocks could cause a problem?

Reply:
Not currently, as footnotes are essentially a top-level block element, but there are other places we're using this method that I am concerned about--I suspect we may need to visit the `close_all_blocks()` method in general if/when the document model becomes more complex/nested.